firmware/defmt-itm/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+//! [`defmt`](https://github.com/knurling-rs/defmt) global logger over RTT.
+//!
+//! NOTE when using this crate it's not possible to use (link to) the `rtt-target` crate
+//!
+//! To use this crate, link to it by importing it somewhere in your project.
+//!
+//! ```
+//! // src/main.rs or src/bin/my-app.rs
+//! use defmt_rtt as _;
+//! ```
+
+#![no_std]
+
+use core::{
+    ptr::NonNull,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+use cortex_m::{interrupt, itm, peripheral::ITM, register};
+
+static ENABLED: AtomicBool = AtomicBool::new(false);
+
+/// Enables defmt logging over the ITM stimulus port 0
+pub fn enable(itm: ITM) {
+    // enable stimulus port 0
+    itm.ter[0].write(1);
+    drop(itm);
+    ENABLED.store(true, Ordering::Relaxed);
+}
+
+#[defmt::global_logger]
+struct Logger;
+
+impl defmt::Write for Logger {
```

Comment:
(one thing I noticed is that this implementation is exactly the same as the one in `defmt-rtt`)

Reply:
I'm assuming you mean the `acquire`/`release` impl further down?

Reply:
yes, the acquire / release implementations look exactly the same